File: app/src/androidTest/java/com/mbmc/template/helper/Matchers.java
package com.mbmc.template.helper;

import android.support.test.espresso.matcher.BoundedMatcher;
import android.text.TextUtils;
import android.view.View;
import android.widget.TextView;

import org.hamcrest.Description;
import org.hamcrest.Matcher;


public final class Matchers {

    public static Matcher<View> isEmpty() {
        return new BoundedMatcher<View, TextView>(TextView.class) {
            @Override
            public boolean matchesSafely(TextView view) {
                return TextUtils.isEmpty(view.getText());
            }

            @Override
            public void describeTo(Description description) {
                description.appendText("TextView text is empty");
            }
        };
    }


    private Matchers() {

    }

}

File: feature/search/src/androidTest/kotlin/com/mbmc/template/feature/search/helper/Matchers.kt
package com.mbmc.template.feature.search.helper

import android.view.View
import android.view.ViewGroup
import org.hamcrest.Description
import org.hamcrest.Matcher
import org.hamcrest.TypeSafeMatcher

// https://stackoverflow.com/questions/24748303/selecting-child-view-at-index-using-espresso/30073528#30073528
fun nthChildOf(parentMatcher: Matcher<View>, childPosition: Int): Matcher<View> {
    return object : TypeSafeMatcher<View>() {
        override fun describeTo(description: Description) {
            description.appendText("with $childPosition child view of type parentMatcher")
        }

        override fun matchesSafely(view: View): Boolean {
            if (view.parent !is ViewGroup) {
                return parentMatcher.matches(view.parent)
            }

            val group = view.parent as ViewGroup
            return parentMatcher.matches(view.parent) && group.getChildAt(childPosition) == view
        }
    }
}